core/src/test/java/nl/nn/adapterframework/stream/MessageTest.java

```diff
@@ -1162,4 +1163,52 @@ public int read(char[] chr, int st, int end) throws IOException {
 
 		assertEquals(charsToRead, charsRead.get());
 	}
+
+	@Test
+	public void testCopyMessage1() throws IOException {
+		// Arrange
+		Message msg1 = Message.asMessage("a");
+
+		// Act
+		Message msg2 = msg1.copyMessage();
+
+		msg1.close();
+
+		// Assert
+		assertNull(msg1.asObject());
+		assertNotNull(msg2.asObject());
+		assertEquals("a", msg2.asString());
+	}
+
+	@Test
+	public void testCopyMessage2() throws IOException {
+		// Arrange
+		Message msg1 = Message.asMessage(new StringReader("á"));
+
+		// Act
+		Message msg2 = msg1.copyMessage();
+
+		msg1.close();
+
+		// Assert
+		assertNull(msg1.asObject());
+		assertNotNull(msg2.asObject());
+		assertEquals("á", msg2.asString());
+	}
+
+	@Test
+	public void testCopyMessage3() throws IOException {
```

Comment:
Ik geloof wel dat dit goed gaat, maar eigenlijk zouden we dit ook met een grote file moeten testen.

Reply:
> Ik geloof wel dat dit goed gaat, maar eigenlijk zouden we dit ook met een grote file moeten testen.

Iedere stream van onbekende size wordt als SerializableFileReference ge-preserved dus zelfde als een grote file, volgens mij?
Daarom heb ik niet geprobeerd om dat apart te testen.